omnibus/config/software/datadog-agent-integrations-py2.rb

```diff
@@ -278,49 +285,92 @@
-      end
+        # Copy the metric file, if it exists
+        metrics_yaml = "#{check_dir}/datadog_checks/#{check}/data/metrics.yaml"
+        if File.exist? metrics_yaml
+          mkdir check_conf_dir
+          copy metrics_yaml, "#{check_conf_dir}/" unless File.exist? "#{check_conf_dir}/metrics.yaml"
+        end
 
-      # Copy SNMP profiles
-      profiles = "#{check_dir}/datadog_checks/#{check}/data/profiles"
-      if File.exist? profiles
-        copy profiles, "#{check_conf_dir}/"
-      end
+        # We don't have auto_conf on windows yet
+        auto_conf_yaml = "#{check_dir}/datadog_checks/#{check}/data/auto_conf.yaml"
+        if File.exist? auto_conf_yaml
+          mkdir check_conf_dir
+          copy auto_conf_yaml, "#{check_conf_dir}/" unless File.exist? "#{check_conf_dir}/auto_conf.yaml"
+        end
 
-      if windows?
-        command "#{python} -m pip wheel . --no-deps --no-index --wheel-dir=#{wheel_build_dir}", :cwd => "#{windows_safe_path(project_dir)}\\#{check}"
-        command "#{python} -m pip install datadog-#{check} --no-deps --no-index --find-links=#{wheel_build_dir}"
-      else
-        command "#{pip} wheel . --no-deps --no-index --wheel-dir=#{wheel_build_dir}", :env => nix_build_env, :cwd => "#{project_dir}/#{check}"
-        command "#{pip} install datadog-#{check} --no-deps --no-index --find-links=#{wheel_build_dir}"
+        # Copy SNMP profiles
+        profiles = "#{check_dir}/datadog_checks/#{check}/data/profiles"
+        if File.exist? profiles
+          copy profiles, "#{check_conf_dir}/"
+        end
+
+        cached_wheel_glob = Dir.glob(File.join(cached_wheels_dir.gsub("\\", "/"), "datadog_#{check}-*.whl"))
+        if cached_wheel_glob.length == 1
+          wheel_path = windows_safe_path(cached_wheel_glob[0])
+          command "#{pip} install --no-deps --no-index #{wheel_path}"
+          next
+        elsif cached_wheel_glob.length > 1
+            raise "Found multiple wheels for #{check}: #{cached_wheel_glob}"
+        end
+
+        if windows?
+          command "#{python} -m pip wheel . --no-deps --no-index --wheel-dir=#{wheel_build_dir}", :cwd => "#{windows_safe_path(project_dir)}\\#{check}"
+          command "#{python} -m pip install datadog-#{check} --no-deps --no-index --find-links=#{wheel_build_dir}"
+        else
+          command "#{pip} wheel . --no-deps --no-index --wheel-dir=#{wheel_build_dir}", :env => nix_build_env, :cwd => "#{project_dir}/#{check}"
+          command "#{pip} install datadog-#{check} --no-deps --no-index --find-links=#{wheel_build_dir}"
+        end
+        if cache_bucket
+          command "inv -e agent.upload-integration-to-cache " \
+            "--python 2 --bucket #{cache_bucket} " \
+            "--integrations-dir #{windows_safe_path(project_dir)} " \
+            "--build-dir #{wheel_build_dir} " \
+            "--integration #{check} " \
+            "--awscli #{awscli}",
+            :cwd => tasks_dir_in
+        end
       end
     end
 
```

Comment:
The lines below (377 to 390) should be in the `block`, otherwise they get run before the contents of the `block` are run, since all the `commands` added in the `block` are put _after_ the `patches`, `commands`, etc. that are defined outside of the block (and doing `pip check` before the integrations are installed isn't very useful).